src/ros2/usb_cam_node.cpp

```diff
@@ -391,6 +391,8 @@ bool UsbCamNode::take_and_send_image_mjpeg()
 
   // grab the image, pass image msg buffer to fill
   m_camera->get_image(reinterpret_cast<char *>(&m_compressed_img_msg->data[0]));
+  m_compressed_img_msg->data.resize(m_camera->get_image_size_in_bytes());
+  m_camera->reset_image_size_in_bytes(); // reset for next grab
```

Comment:
@hilookas I dont think this does what you think it does 😅 the `set_size_in_bytes()` method (called inside the `reset_image_size_in_bytes` method you added above, just changes the `image_t` structure size to whatever the height and `bytes_per_line` values are set to. So if those two values don't change at all, this function wouldn't change anything / reset it to anything.